src/main/java/build/buildfarm/worker/shard/ShardWorkerContext.java

```diff
@@ -122,6 +132,7 @@
   private final boolean limitExecution;
   private final boolean limitGlobalExecution;
   private final boolean onlyMulticoreTests;
+  private final boolean keepInsertFiles;
```

Comment:
I would prefer if this class were given a Write-supplier that did the 'right' thing - there doesn't need to be any functionality split for this class to see.